src/ui/table/Table.jsx

```diff
@@ -1,86 +1,113 @@
-import {TableStyle} from "./Style";
 import {UI} from "../UIBase";
+import {TableStyle} from "./Style";
 import {registerStyle} from "../style/Theme";
 
-// TODO: the whole table architecture probably needs a rethinking
-class TableRow extends UI.Primitive("tr") {
-    render() {
-        let rowCells = [];
-
-        for (let column of this.options.columns) {
-            rowCells.push(this.renderEntryCell(column));
+export class ColumnHandler {
+    constructor(options, index) {
```

Comment:
Cool change. Also, the one that destroys the dashboard and makes this branch necessary :))